log/api_test.go

```diff
@@ -0,0 +1,143 @@
+package log
+
+import (
+	"context"
+	"testing"
+
+	"github.com/anz-bank/pkg/log/loggers"
+	"github.com/anz-bank/pkg/log/testutil"
+	"github.com/marcelocantos/frozen"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestWithLogger(t *testing.T) {
+	t.Parallel()
+
+	logger := loggers.NewMockLogger()
+	logger.On("Copy").Return(logger).Once()
+	WithLogger(context.Background(), logger)
```

Comment:
right, I will put that in a comment, but basically since it uses a mock logger, I can set it to return whatever I want (using the `Return` method in `logger.On("Copy").Return(logger).Once()`) so I thought it's not really useful to check the return value of `WithLogger`. So that test just asserts that the `Copy` method is called.

Reply:
i moved the assertion in a function along with the comment